### ISSUE TITLE
`CloudFrontToApiGatewayToLambda` Deployment Fails with `InvalidBucketAclWithObjectOwnership`

### ISSUE THREAD

**Issue by ComfyDust** [awslabs/aws-solutions-constructs#957]
Starting in April 2023, S3 started setting the `BucketOwnerEnforced` setting on new buckets by default, disabling the use of ACLs (see "ACLs Disabled" in [Heads-Up: Amazon S3 Security Changes Are Coming in April of 2023](https://aws.amazon.com/blogs/aws/heads-up-amazon-s3-security-changes-are-coming-in-april-of-2023/)). As a result, whenever the `CloudFrontToApiGatewayToLambda` construct is deployed for the first time with default settings, the deployment fails with the below error since the `LogDeliveryWrite` ACL [is automatically set](https://github.com/awslabs/aws-solutions-constructs/blob/de47a165dfbd9eb1f57fc734aed46dae776d82ea/source/patterns/%40aws-solutions-constructs/core/lib/s3-bucket-helper.ts#L58).

### Reproduction Steps

Deploy a new instance of `CloudFrontToApiGatewayToLambda` with default settings

### Error Log

>  Bucket cannot have ACLs set with ObjectOwnership's BucketOwnerEnforced setting (Service: Amazon S3; Status Code: 400; Error Code: InvalidBucketAclWithObjectOwnership; Request ID: [...]; S3 Extended Request ID: [...]; Proxy: null)

### Environment

  - **CDK CLI Version      :** 2.67.0
  - **CDK Framework Version:** 2.67.0
  - **AWS Solutions Constructs Version     :** 2.21.0
  - **OS                   :** ubuntu22
  - **Language             :** TypeScript

### Other

Ideally, `CloudFrontToApiGatewayToLambda` should be updated to set `BucketOwnerEnforced` by default. (This probably can't be done in a minor version change though since it changes the previous ownership default of `ObjectWriter`.)

In the meantime, as a workaround, the object ownership setting can be overridden via the `cloudFrontLoggingBucketProps`. Example:

```typescript
new CloudFrontToApiGatewayToLambda(
  this,
  'CloudFrontToApiGatewayToLambda',
  {
    // ...
    cloudFrontLoggingBucketProps: {
      objectOwnership: ObjectOwnership.BUCKET_OWNER_PREFERRED,
    },
  }
)
```

--- 

This is :bug: Bug Report

**Comment by biffgaut** [comment]
This is the result of a change in the S3 service to default ACLs by default. It is fixed in Solutions Constructs as of release 2.38.0.

**Comment by ComfyDust** [comment]
Thanks for confirming the fix! Missed that in the release.
